api/logic/Explanation.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package grakn.client.api.logic;
+
+import grakn.client.api.answer.ConceptMap;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface Explanation {
+
+    Rule rule();
+
+    ConceptMap thenAnswer();
+
+    ConceptMap whenAnswer();
+
+    Map<String, Set<String>> variableMapping();
```

Comment:
What are `thenAnswer`, `whenAnswer` and `variableMapping`?

Reply:
So given a particular `Rule` that is being explained, these are the `ConceptMap` were substituted into the `Rule` when and then respectively, to reach the inference that answered your original answer

Reply:
oh and the `variableMapping` is there because your query isn't necesarily in the same variable space as the rule, so this provides that mapping